hazelcast-openshift-rhel/healthcheck.sh

```diff
@@ -1,3 +1,14 @@
-#!/bin/bash                                                                     
-result=$( curl http://127.0.0.1:5701/hazelcast/healthcheck | awk 'match($0, /Hazelcast\:\:ClusterSafe\=([A-Z]*)/)' )                                            
-if [ "$result" == "Hazelcast::ClusterSafe=TRUE" ]; then exit 0; else exit 1; fi 
+#!/bin/bash
+URL="http://127.0.0.1:5701/hazelcast/healthcheck"
+
+HTTP_RESPONSE=$(curl --silent --write-out "HTTPSTATUS:%{http_code}" $URL)
+HTTP_BODY=$(echo $HTTP_RESPONSE | sed -e 's/HTTPSTATUS\:.*//g')
+HTTP_STATUS=$(echo $HTTP_RESPONSE | tr -d '\n' | sed -e 's/.*HTTPSTATUS://')
+
+if [ ! $HTTP_STATUS -eq 200  ]; then
+  echo "failure, http status: "$HTTP_STATUS
+  exit 1
+fi
+
+result=$(echo $HTTP_BODY | awk '{match($0,  /Hazelcast\:\:NodeState\=([A-Z]*)/, m)}END{print m[1]}')
```

Comment:
I see, however when we add a new pod from OCP, `ClusterSafe` status becomes false, therefore all pods are restarted by Openshift itself, which also causes data loss, that was the best solution in my current understanding, do you have any solution alternatives? can we combine `NodeState` and `ClusterSafe` to get a liveness verdict?